airbyte-db/db-lib/src/main/java/io/airbyte/db/jdbc/AbstractJdbcCompatibleSourceOperations.java

```diff
@@ -146,7 +148,11 @@ protected void putDefault(final ObjectNode node, final String columnName, final
   }
 
   protected void setTime(final PreparedStatement preparedStatement, final int parameterIndex, final String value) throws SQLException {
-    setTimestamp(preparedStatement, parameterIndex, value);
+    try {
+      preparedStatement.setObject(parameterIndex, LocalTime.parse(value));
```

Comment:
> does this handle time with timezone values correctly? E.g. `01:23:45.67Z` (asking because it's using a LocalTime object)
> 
> (or am I misunderstanding how this method is supposed to be used?)

Nope, it's related to `TIME` datatype only and not to`TIMETZ`
I'm wondering that we don't have common mapping in JdbcSourceOperations for TIMETZ and TIMESTAMPTZ datatypes to use it as a cursor field. 
https://github.com/airbytehq/airbyte/blob/master/airbyte-db/db-lib/src/main/java/io/airbyte/db/jdbc/JdbcSourceOperations.java#L66-L90
Currently it's implemented only for Postgres Source

I believe that we should have a separate task to use TIMETZ and TIMESTAMPTZ as a cursor for all JDBC sources

Reply:
got it. Does that mean that even after this PR is merged, `timetz` values (for postgres + cockroachdb) will still be `read` as full iso8601 timestamps?

Reply:
So this is fixing the `setTime` in the base class? If so, can you do the same and delete the redundant methods from inherited classes?

Reply:
> got it. Does that mean that even after this PR is merged, `timetz` values (for postgres + cockroachdb) will still be `read` as full iso8601 timestamps?
> 
> edit: or is this the special handling you added in `putCockroachSpecialDataType`, and postgres was already behaving correctly?

yes, that's right

Reply:
> So this is fixing the `setTime` in the base class? If so, can you do the same and delete the redundant methods from inherited classes?

removed redundant methods from inherited classes for Postgres and MySQL